src/rgw/rgw_lc.h

```diff
@@ -175,13 +213,17 @@ class LCFilter
     return obj_tags;
   }
 
+  const uint32_t get_flags() const {
+    return flags;
+  }
+
   bool empty() const {
-    return !(has_prefix() || has_tags());
+    return !(has_prefix() || has_tags() || has_flags());
   }
 
   // Determine if we need AND tag when creating xml
   bool has_multi_condition() const {
-    if (obj_tags.count() > 1)
+    if (obj_tags.count() + int(has_prefix()) > 1) // Prefix is a member of Filter
```

Comment:
shouldn't it include has_flags() as well? as ArchiveZone flag can be set even with one obj_tag as listed in this example -
https://github.com/ceph/ceph/pull/46928/files#diff-00cdaf79e8833ed31a361d881a37c7434678d84f916789f4e1bc5eab32b25427R50

Reply:
ah, I think so